x-pack/plugins/security_solution/public/detections/components/rules/rule_actions_overflow/index.tsx

```diff
@@ -96,22 +96,24 @@ const RuleActionsOverflowComponent = ({
             <EuiContextMenuItem
               key={i18nActions.EXPORT_RULE}
               icon="exportAction"
-              disabled={userHasNoPermissions || rule.immutable}
+              disabled={!userHasPermissions || rule.immutable}
               data-test-subj="rules-details-export-rule"
-              onClick={() => {
-                setIsPopoverOpen(false);
-                setRulesToExport([rule.rule_id]);
+              onClick={async () => {
+                closePopover();
+                const blob = await exportRules({ ids: [rule.rule_id] });
+                downloadBlob(blob, `${i18nActions.EXPORT_FILENAME}.ndjson`);
+                displaySuccessToast(i18nActions.SUCCESSFULLY_EXPORTED_RULES(1), dispatchToaster);
```

Comment:
The `<GenericDownloader />` component also presented users with an error toaster if something went wrong. Should we handle potential errors here too?

Reply:
You are right, thank you for the suggestion 👍
I'll add a try-catch block to handle errors properly.